File: linear-search/node/LinearSearch.js
/**
 * Linear search algorithm implementation.
 * @param {Array} array Array of numbers.
 * @param {Number} number Number to be searched.
 * @returns {Number} Number index, if not found returns -1.
 */
const linearSearch = (array, number) => {
    for (let index in array) {
        if (array[index] === number) {
            return index;
        }
    }

    return -1;
}

module.exports = { linearSearch };

File: linear-search/java/src/LinearSearch.java
/**
 * This class has an Static method who implements Linear Search.
 * @author JohanCamiloL
 */
public class LinearSearch {

    /**
     * Linear search on array number elements
     *
     * @param array  array with 0 or more numbers.
     * @param number number to be searched
     * @return Index of the element to search, if not found, returns -1.
     */
    public static int linearSearch(int[] array, int number) {
        for (int index = 0; index < array.length; index++) {
            if (array[index] == number) {
                return index;
            }
        }

        return -1;
    }

}
